src/lightning/pytorch/strategies/fsdp.py

```diff
@@ -440,6 +440,43 @@ def register_strategies(cls, strategy_registry: _StrategyRegistry) -> None:
         )
         cls._registered_strategies.append("fsdp_cpu_offload")
 
+    def lightning_module_state_dict(self) -> Dict[str, Any]:
+        assert self.model is not None
+        if self._state_dict_type == "sharded":
+            state_dict_ctx = _get_sharded_state_dict_context(self.model)
+        elif self._state_dict_type == "full":
+            state_dict_ctx = _get_full_state_dict_context(self.model, offload_to_cpu=(self.world_size > 1))
```

Comment:
Does Fabric also need this `world_size > 1` logic? I don't find it there

Reply:
I couldn't replicate an issue with world_size=1 for FSDP and offload_to_cpu=True so I'm not sure. Leaving it as is for now.

Reply:
Thats strange. What's the issue? It would be good to describe it here for posterity, even if left as is for now

Reply:
Ah ok, it is only an issue in torch 2.0. I'll make sure we standardize this across both PyTorch and Fabric in #18371.